.ci/static-checks.sh

```diff
@@ -284,13 +314,49 @@ check_docs()
 
 	info "Checking documentation"
 
-	local docs=$(find . -name "*.md" |grep -v "vendor/" || true)
-
+	local doc
+	local docs
+	local docs_status
+	local new_docs
+	local new_urls
 	local url
+
+	if [ "$master_branch" = "true" ]
+	then
+		# Check all documents
+		docs=$(find . -name "*.md" |grep -v "vendor/" || true)
+	else
+		# Check changed documents only
+		docs_status=$(get_pr_changed_file_details || true)
+		docs_status=$(echo "$docs_status"|grep "\.md$" || true)
+
+		docs=$(echo "$docs_status"|awk '{print $NF}')
+
+		# Newly-added docs
+		new_docs=$(echo "$docs_status"|grep "^A"|awk '{print $NF}')
```

Comment:
You could probably combine that grep into the awk itself with a `/^A/ {print $NF}` maybe.

Reply:
Good catch - fixed.